DnD-Montreal/session-tome
# Add BulkEntryController endpoint

## Description
Closes #115 
Adds a controller endpoint that accepts a date range and creates the specified number of entries within that date range.

Endpoint: POST ➡️  `/entry-bulk`
Accepts: 
- `character_id`: Id of the character to add the entries too
- `adventure_id`: the adventure played for all of the entries
- `start_date`: the day of the first session played
- `end_date`: the day of the last session played
- `frequency`: how many time per week a game was played (1 -> once per week, 2 -> twice per week, 0.5 -> once every 2 weeks)

## How to test
Note: there is now a basic seeder that generates some sensible data for you, to get a fresh database before testing you can go ahead and run `php artisan migrate:fresh && php artisan db:seed`

1. Generate a Bearer token (Tinker: `User::first()->createToken('test')`)
2. In Insomnia > Auth > Bearer Token, place the `plainTextToken` from tinker in the token field
3. Fill the form similarly to the attached image, feel free to try different configurations of parameters
4. Send the request and observe there are more entries than before.

## Additional Context
How my Insomnia request is filled:
![image](https://user-images.githubusercontent.com/9440691/139335919-fd1582bb-b0ad-4499-ba4a-b379c25547db.png)
